aptos-move/block-executor/src/view.rs

```diff
@@ -821,6 +824,142 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
             .collect()
     }
 
+    fn get_group_reads_needing_exchange_parallel(
+        &self,
+        parallel_state: &ParallelState<'a, T, X>,
+        delayed_write_set_keys: &HashSet<T::Identifier>,
+        skip: &HashSet<T::Key>,
+    ) -> Result<
+        BTreeMap<
+            T::Key,
+            (
+                T::Value,
+                Vec<(T::Tag, (T::Value, Option<Arc<MoveTypeLayout>>))>,
+                u64,
+            ),
+        >,
+        PanicError,
+    > {
+        parallel_state
+            .captured_reads
+            .borrow()
+            .get_group_read_values_with_delayed_fields()
+            .filter(|(key, _)| !skip.contains(key))
+            .flat_map(|(key, group_read)| {
+                let GroupRead { inner_reads, .. } = group_read;
+                let mut resources_needing_delayed_field_exchange = vec![];
+                for (tag, data_read) in inner_reads {
+                    if let DataRead::Versioned(_version, value, Some(layout)) = data_read {
+                        if let Some(bytes) = value.bytes() {
+                            let identifiers_in_read = self
+                                .extract_identifiers_from_value(bytes, layout.as_ref())
+                                .unwrap();
+                            if !delayed_write_set_keys.is_disjoint(&identifiers_in_read) {
+                                // TODO[agg_v2](optimize): Is it possible to avoid clones here?
```

Comment:
these are clones of Arc's, which should be fine? (except for the key maybe)

Reply:
We are doing .as_ref().clone(). Is that still fine?

Reply:
Bytes are basically Arc internally, so we have Arc<Arc<>>, and are cloning inner Arc